Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/FlexPanel.cs

```diff
@@ -0,0 +1,1613 @@
+                        // attribute values. We are using a simple insertion sort as we need
+                        // stability (insertion order must be preserved) and cross-platform
+                        // support. We should eventually switch to merge sort (or something
+                        // else) if the number of items becomes significant enough.
+                        for (int i = 0; i < item.Count; i++)
+                        {
+                            indices[i] = i;
+                            for (int j = i; j > 0; j--)
+                            {
+                                int prev = indices[j - 1];
+                                int curr = indices[j];
+                                if (item.Children[prev].Order <= item.Children[curr].Order)
+                                {
+                                    break;
+                                }
+
+                                indices[j - 1] = curr;
+                                indices[j] = prev;
+                            }
+                        }
+
+                        _ordered_indices = indices;
+                    }
+
+                    Flex_dim = 0;
+                    Flex_grows = 0;
+                    Flex_shrinks = 0;
+
+                    Reverse2 = false;
+                    Wrap = item.Wrap != FlexWrap.NoWrap;
+                    if (Wrap)
+                    {
+                        if (item.Wrap == FlexWrap.Reverse)
+                        {
+                            Reverse2 = true;
+                            Pos2 = Align_dim;
+                        }
+                    }
+                    else
+                    {
+                        Pos2 = Vertical ? item.PaddingLeft : item.PaddingTop;
+                    }
+
+                    Need_lines = Wrap && item.AlignContent != FlexAlignContent.Start;
+                    Lines = null;
+                    Lines_sizes = 0;
+                }
+
+                public FlexItem Child_at(FlexItem item, int i) =>
+                    item.Children[_ordered_indices?[i] ?? i];
+
+                public void Cleanup()
+                {
+                    _ordered_indices = null;
+                    Lines = null;
+                }
+            }
+
+            IEnumerator IEnumerable.GetEnumerator() =>
+                ((IEnumerable<FlexItem>)this).GetEnumerator();
+
+            IEnumerator<FlexItem> IEnumerable<FlexItem>.GetEnumerator() =>
+                (Children ?? System.Linq.Enumerable.Empty<FlexItem>()).GetEnumerator();
+
+            private void ValidateChild(FlexItem child)
+            {
+                if (this == child)
+                {
+                    throw new ArgumentException("cannot add item into self");
+                }
+
+                if (child.Parent != null)
+                {
+                    throw new ArgumentException("child already has a parent");
+                }
```

Comment:
Same as above, this should be using either `Guard` or `ThrowHelper` for better codegen.

Reply:
Why am I just now learning about Guard!  THANK YOU!

Reply:
Ahahahah that's because `Guard` has only been available since the 6.1 release which was published in July, so just for 2 months now, and I've added the `ThrowHelper` APIs with a recent PRs just a few weeks ago so they're not yet available on NuGet at all, it's all pretty new. Really happy to hear you like them though, that's great! 😄